### PR TITLE
Don't hardcode the use of https for X-Forwarded-Proto

### PR DESCRIPTION
**Pull request by FlorianSW**
Based on the X-Forwarded-Proto header, openHAB will decide, what protocol is used for url's porvided in the output of the REST API request. Hardcofing it to https forces hosters of the openhab-cloud application to use https, otherwise a connection of REST based applications (e.g. the openHAB app) will not work (trying to connect using https to a http only page).

This commit dynamically determines, what protocol should be used for the X-Forwarded-Proto header and sets it automatically. The preference is set to be the protocol of the openhab-cloud service url (baseURL), which is configured in the plugin. If the baseURL could not be parsed (normally, the service shouldn't be reachable in this case), it will check, if the openHAB-cloud app provides the protocol in the request data and, as a fallback, https will be used, if both other methods failed.

See also: https://github.com/openhab/openhab-cloud/issues/88

### CODE REVIEW

File: addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java [hunk 1]
```diff
@@ -164,6 +171,12 @@ public void connect() {
         } catch (URISyntaxException e) {
             logger.error("Error creating Socket.IO: {}", e.getMessage());
         }
+        try {
```
Comment (martinvw) [inline addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: In my opinion you should collapse the two try-catch blocks, there goal is to separate the error handling from the flow of code, you could also consider to extract the try-catch to a seperate method maybe that is even better.

File: addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java [hunk 2]
```diff
@@ -164,6 +171,12 @@ public void connect() {
...
+            URL parsed = new URL(baseURL);
+            protocol = parsed.getProtocol();
+        } catch (MalformedURLException e) {
+            logger.error("Error parsing baseURL to get protocol, assuming http. Error: {}", e.getMessage());
```
Comment (martinvw) [inline addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: > assuming http

It seems that you fallback to `"https"` on line 351

File: addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java [hunk 3]
```diff
@@ -330,7 +343,14 @@ private void handleRequestEvent(JSONObject data) {
             logger.debug("Request method is {}", requestMethod);
             Request request = jettyClient.newRequest(requestUri);
             setRequestHeaders(request, requestHeadersJson);
-            request.header("X-Forwarded-Proto", "https");
+            if (protocol != null) {
+                request.header("X-Forwarded-Proto", protocol);
```
Comment (martinvw) [inline addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: You could consider not duplicating the `request.header("X-Forwarded-Proto"` 3 times, but just collecting a string and passing that to the `request.header("X-Forwarded-Proto"`

File: addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java [hunk 4]
```diff
@@ -330,7 +343,14 @@ private void handleRequestEvent(JSONObject data) {
...
+            } else if (data.has("protocol")) {
+                request.header("X-Forwarded-Proto", data.getString("protocol"));
```
Comment (martinvw) [inline addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: How did you decide the order, why is the protocol from the base url the most important and then this one? If you would give the `data.getString('protocol')` precendence your code could be simpler, by just setting the protocol to https in the catch block.

File: addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java (hunk 1 above)
Comment (FlorianSW) [inline addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: You're right, I'll do that!

File: addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java (hunk 2 above)
Comment (FlorianSW) [inline addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: changed, my fault.

File: addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java (hunk 3 above)
Comment (FlorianSW) [inline addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: done!

File: addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java (hunk 4 above)
Comment (FlorianSW) [inline addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java]: The protocol of the request should have the preference, my fault.

**Comment by martinvw** [comment]
LGTM thanks for the quick update

**Comment by FlorianSW** [comment]
Thanks! :)

**Pull request merged**